html/includes/forms/update-ifspeed.inc.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2014 Neil Lathwood <https://github.com/laf/ http://www.lathwood.co.uk/fa>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+*/
+header('Content-type: application/json');
+
+$status  = 'error';
+
+$speed = mres($_POST['speed']);
+$device_id = mres($_POST['device_id']);
+$ifName = mres($_POST['ifName']);
+$port_id = mres($_POST['port_id']);
+
+if (!empty($ifName) && is_numeric($port_id) && is_numeric($port_id)) {
+    // We have ifName and  port id so update ifAlias
+    if (empty($speed)) {
+        $speed = 999999;
+        $high_speed = 999999;
+        // Set to 999999 so we avoid using ifDescr on port poll
+    }
+    else {
+        $high_speed = $speed / 1000000;
+    }
+    if (dbUpdate(array('ifSpeed'=>$speed, 'ifHighSpeed'=>$high_speed), 'ports', '`port_id`=?', array($port_id)) > 0) {
```

Comment:
You can detect unchanged values in JavaScript by storing the previous value in a data attribute.